trace-server/org.eclipse.tracecompass.incubator.trace.server.jersey.rest.core/src/org/eclipse/tracecompass/incubator/internal/trace/server/jersey/rest/core/services/DataProviderService.java

```diff
@@ -1129,6 +1148,247 @@ public Response getStyles(
+                .setCategory(outputId).build()) {
+            TmfExperiment experiment = ExperimentManagerService.getExperimentByUUID(expUUID);
+            if (experiment == null) {
+                return Response.status(Status.NOT_FOUND).entity(NO_SUCH_TRACE).build();
+            }
+
+            Response errorResponse = validateOutputConfigParameters(outputId, queryParameters);
+            if (errorResponse != null) {
+                return errorResponse;
+            }
+
+            IDataProviderFactory factory = manager.getFactory(outputId);
+            if (factory == null) {
+                return Response.status(Status.METHOD_NOT_ALLOWED).entity(NO_SUCH_PROVIDER).build();
+            }
+
+            String typeId = queryParameters.getTypeId();
+
+            ITmfDataProviderSource source = factory.getAdapter(ITmfDataProviderSource.class);
+
+            if (source == null) {
+                return Response.status(Status.METHOD_NOT_ALLOWED).entity(NO_SUCH_PROVIDER).build();
+            }
+
+            Optional<ITmfConfigurationSourceType> optional = source.getConfigurationSourceTypes().stream().filter(type -> type.getId().equals(typeId)).findAny();
+            if (!optional.isPresent()) {
+                return Response.status(Status.NOT_FOUND).entity("Configuration source type doesn't exist").build(); //$NON-NLS-1$
+            }
+            ITmfConfigurationSourceType sourceType = optional.get();
+//            @NonNull Map<@NonNull String, @NonNull Object> params = new HashMap<>();
+//            if (sourceType.getSchemaFile() != null) {
+//                // Pass JSON object as json string
+//                params.put(TmfConfiguration.JSON_STRING_KEY, queryParameters.getParameters().toString());
+//            } else {
+//                // Convert it to a Map<String, Object>
+//                ObjectMapper mapper = new ObjectMapper();
+//                params = mapper.convertValue(queryParameters.getParameters(), new TypeReference<Map<String,Object>>(){});
+//            }
+            List<IDataProviderDescriptor> returnDescr = source.createDataProviderDescriptors(typeId, experiment, queryParameters.getParameters().toString());
+            return Response.ok(returnDescr).build();
+        } catch (TmfConfigurationException e) {
+            return Response.status(Status.BAD_REQUEST).entity(e.getMessage()).build();
+        }
+    }
+
+    /**
+     * DELETE a derived data provider from created by a given data provider
+     *
+     * @param expUUID
+     *            desired experiment UUID
+     * @param outputId
+     *            Output ID for the data provider to query
+     * @param derivedOutputId
+     *            Output ID for the data provider to delete
+     * @return status and the deleted configuration instance, if successful
+     */
+    @DELETE
+    @Path("/{outputId}/{derivedOutputId}")
```

Comment:
Should we change the following? This would require that the derived output know how to delete it.
```
@Path("/{derivedOutputId}/")
```

As proposed earlier (https://github.com/eclipse-tracecompass-incubator/org.eclipse.tracecompass.incubator/pull/88#issuecomment-2399802653) to have a way to remove a config from the data provider instance which will remove the actual data provider(s) consequently.

@abhinava-ericsson @PatrickTasse WDYT?

Reply:
I would say  `@Path("/{outputId}/{derivedOutputId}")` is cleaner. In general, it sounds like a nice idea to have all url references to derived output ids under a base output id.